build.gradle

```diff
@@ -51,6 +51,11 @@ tasks.withType(Test).configureEach {
     def testGradleVersion = findProperty('testGradleVersion') ?: GradleVersion.current().version
     systemProperty 'testContext.gradleVersion', testGradleVersion
     project.develocity.buildScan.value(identityPath.path + "#gradleVersion", testGradleVersion)
+
+    def javaHome = javaToolchains.launcherFor {
+        languageVersion = JavaLanguageVersion.of(8)
+    }.map { it.metadata.installationPath }
+    systemProperty 'testContext.javaHome', javaHome.get().asFile.absolutePath
```

Comment:
I'd use a [`CommandLineArgumentProvider`](https://docs.gradle.org/current/userguide/caching_java_projects.html#dealing_with_file_paths) with an `@Input` for the Java major version and the actual path marked as `@Internal`. Otherwise, this will cause cache misses between machines.

Reply:
🤦  Yes, ofc.